### ISSUE TITLE
ConfigMappings with single List of objects can't be processed

### ISSUE THREAD

**Issue by ruromero** [smallrye/smallrye-config#517]
After some testing with what was added in https://github.com/smallrye/smallrye-config/pull/489
And starting with the example provided here: https://github.com/quarkusio/quarkus/blob/master/integration-tests/smallrye-config/src/main/java/io/quarkus/it/smallrye/config/Server.java

The following example doesn't seem to work:
```
@ConfigMapping(prefix = "server")
public interface Server {
        List<Origin> origins();
        interface Origin {
            String host();
            int port();
        }
}
```

Fails with the following error: `server.origins does not map to any root. `
However if I add the `List<String> methods()` it works.

Thanks in advance.

**Comment by gastaldi** [comment]
@ruromero what's in your `application.properties`?

**Comment by radcortez** [comment]
@ruromero I believe the issue is caused by and old version of the yaml config source. You had to upgrade `smallrye-config` in Quarkus `1.12` to be able to use the indexed properties mapping, but the `YamlConfigSource` also had changes to support it, so you need to update that dependency as well: 

```xml
<dependency>
    <groupId>io.smallrye.config</groupId>
    <artifactId>smallrye-config-source-yaml</artifactId>
    <version>1.11.0</version>
</dependency>
```

But I did found an issue with the `@WithParentName` which is not working correctly with `Collections`, so I've added a fix for it. You are going to need to not use it for now. Sorry for the inconvenience.

**Comment by ruromero** [comment]
@radcortez after adding to quarkus 1.12.1.Final the following dependencies:
```
        <dependency>
            <groupId>io.smallrye.config</groupId>
            <artifactId>smallrye-config</artifactId>
            <version>1.11.0</version>
        </dependency>
        <dependency>
            <groupId>io.smallrye.config</groupId>
            <artifactId>smallrye-config-source-yaml</artifactId>
            <version>1.11.0</version>
        </dependency>
```

The error is different, maybe is what you are trying to fix in #519 
```
Caused by: io.smallrye.config.ConfigValidationException: Configuration validation failed:
	servers[0].host does not map to any root
	servers[1].password does not map to any root
	servers[0].password does not map to any root
	servers[0].username does not map to any root
	servers[1].host does not map to any root
	servers[1].username does not map to any root
```

@gastaldi this is the content of the application.yaml file
```
servers:
  - host: http://kieserver1.example.com:8080/kie-server/services/rest/server
    username: krisv
    password: krisv
  - host: http://kieserver2.example.com:8080/kie-server/services/rest/server
    username: admin
    password: admin123
```

**Comment by radcortez** [comment]
@ruromero yes, there is an issue with `@WithParentName` being ignored, fixed by #519. Please, don't use it for now, and it should be fine. I'm sorry for the inconvenience. I'll try to push this to Quarkus ASAP.
